controller/two_lane_queue.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"k8s.io/client-go/util/workqueue"
+)
+
+// item wraps workqueue.Interface.Get() return
+type item struct {
+	i  interface{}
+	sd bool
```

Comment:
This struct is overkill because `sd` is always false.  When it would be true we close the channel, which is actually a better signal overall, so why not just use `chan interface{}`?

Reply:
I guess from the original version. removed.